event_sale_session/static/src/js/EventConfiguratorFormController.esm.js

```diff
@@ -5,32 +5,33 @@
     @author Iván Todorovich <ivan.todorovich@gmail.com>
     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
 */
-import {RelationalModel} from "@web/views/relational_model";
+import {EventConfiguratorRelationalModel} from "@event_sale/js/event_configurator_model";
 import {patch} from "@web/core/utils/patch";
 
-patch(RelationalModel.prototype, "event_sale_session.RelationalModel", {
-    setup(params, {action}) {
-        // Trick to keep model as the main EventConfiguratorRelationalModel is not exported
-        // https://github.com/odoo/odoo/blob/fa142764b2f583793047f794bbc102e6e0807c31/addons/event_sale/static/src/js/event_configurator_model.js#L17
-        // TODO: Try to patch directly the EventConfiguratorRelationalModel
-        this._super(...arguments);
-        this.oldDoAction = action.doAction;
-        const model = this;
-        async function EventConfiguratorDoAction(actionRequest, options = {}) {
-            const state = model.root.data;
-            if (
-                state &&
-                state.event_use_sessions &&
-                actionRequest.type === "ir.actions.act_window_close" &&
-                actionRequest.infos.eventConfiguration &&
-                !options.skip_definition
-            ) {
-                actionRequest.infos.eventConfiguration.event_session_id =
-                    state.event_session_id;
-                options.skip_definition = true;
+patch(
+    EventConfiguratorRelationalModel.prototype,
+    "event_sale_session.EventConfiguratorRelationalModel",
+    {
+        setup(params, {action}) {
+            this._super(...arguments);
+            this.oldDoAction = action.doAction;
+            const model = this;
+            async function EventConfiguratorDoAction(actionRequest, options = {}) {
+                const state = model.root.data;
+                if (
+                    state &&
+                    state.event_use_sessions &&
+                    actionRequest.type === "ir.actions.act_window_close" &&
+                    actionRequest.infos.eventConfiguration &&
+                    !options.skip_definition
+                ) {
+                    actionRequest.infos.eventConfiguration.event_session_id =
+                        state.event_session_id;
+                    options.skip_definition = true;
+                }
+                return model.oldDoAction(actionRequest, options);
             }
-            return model.oldDoAction(actionRequest, options);
-        }
-        this.action.doAction = EventConfiguratorDoAction;
-    },
-});
+            this.action.doAction = EventConfiguratorDoAction;
+        },
```

Comment:
This is what I normally do on cases like this:

```js
patch(
    EventConfiguratorRelationalModel.prototype,
    "event_sale_session.EventConfiguratorRelationalModel",
    {
    /**
     * @override
     */
    async save() {
        const doAction = this.model.doAction;
        this.model.doAction = (actionRequest, options = {}) => {
            actionRequest.infos.eventConfiguration.event_session_id = this.data.event_session_id;
            return doAction(actionRequest, options);
        }
        const res = this._super.apply(this, arguments);
        this.model.doAction = doAction;
        return res;
    }
}
```

Basically, temporarily monkey-patching the method

Reply:
Let me try it :smile: thanks